var/spack/repos/builtin/packages/py-antlr4-python3-runtime/package.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyAntlr4Python3Runtime(PythonPackage):
+    """ANTLR (ANother Tool for Language Recognition) is a powerful
+parser generator for reading, processing, executing, or translating
+structured text or binary files.
+This package provides an ANTLR interface for python3."""
+
+    homepage = "https://www.antlr.org"
+    url      = "https://pypi.io/packages/source/a/antlr4-python3-runtime/antlr4-python3-runtime-4.7.2.tar.gz"
+
+    version('4.7.2', sha256='168cdcec8fb9152e84a87ca6fd261b3d54c8f6358f42ab3b813b14a7193bb50b')
+
+    depends_on('python@3:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
```

Comment:
Why isn't ANTLR listed as a dependency?

Reply:
Good question @citibeth . It seems that the latest antlr4 is a java package, and the runtime support for different languages is distributed in separate packages. This PR provides runtime support for python3. For more information, please see:

https://www.antlr.org/download.html

I suppose we could provide an antlr4 package for spack that includes the parser generator as a core component, with runtime support for different languages as variants. But that is more work than I originally had in mind.

Reply:
I see... so this package is not the typical "Python interface to ANTLR."  Rather, it's stuff needed to run ANTLR-generated Python programs?

Reply:
That's right. I misunderstood the scope of the package myself, hence the misleading description in this PR. I have edited the description, and hopefully the purpose of the package is clearer now.

Reply:
I think the description is still misleading :(.  How about "This package provides runtime libraries required to use Python language ANTLR-generated parsers."

Reply:
I have updated the description again. I think it makes more sense now.

Reply:
I see the doctoring was indented but not updated.  IT does not make sense to me.  To me, I would think it means this is a Python library that allows you to call the ANTLR library.  Like `py-netcdf`, `py-proj`, etc.  There are tons of these "Python interface to something else" libraries.  Because this is NOT such, you need the wording to be more explicit, or else people will assume.

Reply:
My apologies @citibeth ! I changed the description at the top of this PR, but not in the docstring. Please see 2e64ffa .

Reply:
Oops - and 82e4d777 .